### PR TITLE
webui: top devices widget enhancement

### PR DESCRIPTION
**Pull request by crcro**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

added support for: 
- custom widget title
- top devices by: traffic, uptime, response time, cpu usage, memory usage

- fixes: #4447

### CODE REVIEW

File: html/includes/common/top-devices.inc.php [hunk 1]
```diff
@@ -17,51 +18,109 @@
...
+            </div>
+        </div>
+        
+        <div class="form-group">
+            <div class="col-sm-4">
+                <label for="top_query" class="control-label availability-map-widget-header">Top query</label>
+            </div>
+            <div class="col-sm-6">
+                <select class="form-control" name="top_query">
+                    <option value="0" ' . $selected_traffic . '>Traffic</option>
```
Comment (laf) [inline html/includes/common/top-devices.inc.php]: I'd change this to be descriptive values so you don't have to visit this part of the code when checking what the switch call is doing.

File: html/includes/common/top-devices.inc.php [hunk 2]
```diff
@@ -85,46 +144,113 @@
...
+            <tr>
+                <th class="text-left">Device</th>
+                <th class="text-left">' . $table_header . '</a></th>
+            </tr>
+        </thead>
+        <tbody>';
 
     foreach (dbFetchRows($query, $params) as $result) {
+        if ($top_query == 0) {
+            $graphs = generate_device_link($result, generate_minigraph_image($result, $config['time']['day'], $config['time']['now'], 'device_bits', 'no', 150, 21), array(), 0, 0, 0);
```
Comment (laf) [inline html/includes/common/top-devices.inc.php]: You could just set the relevant variables in these if checks and then run generate_minigraph_image at the end.

File: html/includes/common/top-devices.inc.php [hunk 3]
```diff
@@ -85,46 +144,113 @@
...
+        <thead>
+            <tr>
+                <th class="text-left">Device</th>
+                <th class="text-left">' . $table_header . '</a></th>
+            </tr>
+        </thead>
+        <tbody>';
 
     foreach (dbFetchRows($query, $params) as $result) {
+        if ($top_query == 0) {
```
Comment (laf) [inline html/includes/common/top-devices.inc.php]: When the variables are updated, you can use === 'something' to be a strict check.

**Comment by crcro** [comment]
@laf ... hope that now is better :)

**Pull request merged**